orttraining/orttraining/test/python/orttraining_test_ortmodule_execution_agents.py

```diff
@@ -0,0 +1,112 @@
+import numpy as np
+from onnx import TensorProto
+from onnx import helper
+
+import torch
+
+import onnxruntime
+
+from onnxruntime.training.ortmodule._execution_agent import (
+    InferenceAgent,
+    TrainingAgent,
+)
+
+
+def make_inference_model():
+    inputs = [
+        helper.make_tensor_value_info("x", TensorProto.FLOAT, [2, 3]),
+        helper.make_tensor_value_info("y", TensorProto.FLOAT, [2, 3]),
+        helper.make_tensor_value_info("bias", TensorProto.FLOAT, [3]),
+    ]
+    mul_node = helper.make_node("Mul", ["x", "y"], ["mul"])
+    add_node = helper.make_node("Add", ["mul", "bias"], ["add"])
+    outputs = [helper.make_tensor_value_info("add", TensorProto.FLOAT, [2, 3])]
+    graph_def = helper.make_graph([mul_node, add_node], "graph", inputs, outputs)
+    return helper.make_model(graph_def)
+
+
+def make_training_model():
+    inputs = [
+        helper.make_tensor_value_info("x", TensorProto.FLOAT, [2, 3]),
+        helper.make_tensor_value_info("y", TensorProto.FLOAT, [2, 3]),
+        helper.make_tensor_value_info("bias", TensorProto.FLOAT, [3]),
+    ]
+    mul_node = helper.make_node("Mul", ["x", "y"], ["mul"])
+    add_node = helper.make_node("Add", ["mul", "bias"], ["add"])
+    yield_node = helper.make_node(
+        "YieldOp", ["add"], ["add_grad"], full_shape_outputs=[0], domain="com.microsoft"
+    )
+    x_grad_node = helper.make_node("Mul", ["add_grad", "y"], ["x_grad"])
+    y_grad_node = helper.make_node("Mul", ["x", "add_grad"], ["y_grad"])
+    axes_tensor = helper.make_tensor("axes", TensorProto.INT64, [1], [0])
+    bias_grad_node = helper.make_node(
+        "ReduceSum", ["add_grad", "axes"], ["bias_grad"], keepdims=0
+    )
+    outputs = [
+        helper.make_tensor_value_info("x_grad", TensorProto.FLOAT, [2, 3]),
+        helper.make_tensor_value_info("y_grad", TensorProto.FLOAT, [2, 3]),
+        helper.make_tensor_value_info("bias_grad", TensorProto.FLOAT, [3]),
+    ]
+    graph_def = helper.make_graph(
+        [mul_node, add_node, yield_node, x_grad_node, y_grad_node, bias_grad_node],
+        "graph",
+        inputs,
+        outputs,
+        initializer=[axes_tensor],
+    )
+    return helper.make_model(graph_def)
```

Comment:
Maybe move to https://github.com/microsoft/onnxruntime/blob/master/orttraining/orttraining/test/python/_test_commons.py? Maybe wrap on a class with methods to return both model and data